app/api/link-preview/route.ts

```
@@ -0,0 +1,215 @@
+function extractFavicon(html: string, baseUrl: string): string {
+  try {
+    // Try to find favicon link in the HTML
+    const match =
+      /<link\s+rel=["'](?:shortcut )?icon["']\s+href=["']([^"']*)["']/i.exec(
+        html
+      ) ||
+      /<link\s+href=["']([^"']*)["']\s+rel=["'](?:shortcut )?icon["']/i.exec(
+        html
+      );
+
+    if (match && match[1]) {
```

Biome (1.9.4)

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)